optimum/habana/transformers/generation/utils.py

```diff
@@ -1403,14 +1414,26 @@ def greedy_search(
 
             hpu_graphs_kwargs = self._get_hpu_graphs_kwargs(model_kwargs)
 
-            # forward pass to get next token
-            outputs = self(
-                **model_inputs,
-                return_dict=True,
-                output_attentions=output_attentions,
-                output_hidden_states=output_hidden_states,
-                **hpu_graphs_kwargs,
-            )
+            if torch_compile:
```

Comment:
Not even sure we should do it in `generate` at all. If using the trainer, it should already be taken care of (see discussion above). Otherwise, for example in the text-generation example, I think we should just have a `get_torch_compiled_model` in `text-generation/utils.py`. That seems to be the way recommended by Transformers.

Reply:
@regisss thanks for your comments, we will check if we can go with adding get_torch_compiled_model in text-generation/utils.py

Reply:
Ok. I would create 'get_torch_compiled_model' in text-generation/utils.py.